File: hibernate-envers/src/main/java/com/neohoon/enverstest/app/domain/Book.java
package com.neohoon.enverstest.app.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.envers.Audited;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Entity
@Table(name = "book")
@EntityListeners(AuditingEntityListener.class)
@Audited
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class Book extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "book_id", nullable = false, updatable = false)
    private Long id;

    @Column(length = 100)
    private String title;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id")
    private Member author;

    public Book(String title, Member author) {
        this.title = title;
        this.author = author;
    }
}

File: querydsl/src/main/kotlin/com/neohoon/querydsl/app/domain/Book.kt
package com.neohoon.querydsl.app.domain

import jakarta.persistence.*

@Entity
@Table(name = "book")
class Book(
    val name: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id")
    val author: Member
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "book_id", nullable = false, updatable = false)
    val id: Long? = null

}